File: chat/internal/models/message.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Message 消息模型
type Message struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	ChatID    uint           `gorm:"not null" json:"chat_id"`
	SenderID  *uint          `gorm:"index" json:"sender_id,omitempty"`
	Type      string         `gorm:"type:enum('text','document','image','system','ai_assistant');default:'text'" json:"type"`
	Content   *string        `gorm:"type:text" json:"content,omitempty"`
	FileURL   *string        `gorm:"type:varchar(500)" json:"file_url,omitempty"`
	FileName  *string        `gorm:"type:varchar(255)" json:"file_name,omitempty"`
	FileSize  *int64         `gorm:"type:bigint" json:"file_size,omitempty"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	// 关联关系
	Chat      Chat            `gorm:"foreignKey:ChatID" json:"chat,omitempty"`
	Sender    *User           `gorm:"foreignKey:SenderID" json:"sender,omitempty"`
	Statuses  []MessageStatus `gorm:"foreignKey:MessageID" json:"statuses,omitempty"`
	ChatFiles []ChatFile      `gorm:"foreignKey:MessageID" json:"chat_files,omitempty"`
}

// TableName 指定表名
func (Message) TableName() string {
	return "messages"
}

// MessageStatus 消息状态模型
type MessageStatus struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	MessageID uint      `gorm:"not null" json:"message_id"`
	UserID    uint      `gorm:"not null" json:"user_id"`
	Status    string    `gorm:"type:enum('sent','delivered','read','failed');default:'sent'" json:"status"`
	UpdatedAt time.Time `json:"updated_at"`

	// 关联关系
	Message Message `gorm:"foreignKey:MessageID" json:"message,omitempty"`
	User    User    `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

// TableName 指定表名
func (MessageStatus) TableName() string {
	return "message_status"
}

File: laravel/app/Models/Message.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Message extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'chat_id',
        'sender_id',
        'type',
        'content',
        'file_url',
        'file_name',
        'file_size',
    ];

    protected $casts = [
        'file_size' => 'integer',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    /**
     * 所属聊天
     */
    public function chat()
    {
        return $this->belongsTo(Chat::class, 'chat_id');
    }

    /**
     * 发送者
     */
    public function sender()
    {
        return $this->belongsTo(User::class, 'sender_id');
    }

    /**
     * 消息状态
     */
    public function statuses()
    {
        return $this->hasMany(MessageStatus::class, 'message_id');
    }

    /**
     * 获取文件大小（格式化）
     */
    public function getFileSizeFormattedAttribute()
    {
        if (!$this->file_size) return null;
        
        $units = ['B', 'KB', 'MB', 'GB'];
        $size = $this->file_size;
        $unit = 0;
        
        while ($size >= 1024 && $unit < count($units) - 1) {
            $size /= 1024;
            $unit++;
        }
        
        return round($size, 2) . ' ' . $units[$unit];
    }

    /**
     * 是否为系统消息
     */
    public function isSystemMessage()
    {
        return $this->type === 'system';
    }

    /**
     * 是否为文件消息
     */
    public function isFileMessage()
    {
        return in_array($this->type, ['document', 'image', 'video']);
    }
}




